src/main/java/io/sc3/plethora/mixin/EntityMixin.java

```diff
@@ -0,0 +1,26 @@
+package io.sc3.plethora.mixin;
+
+import io.sc3.plethora.util.VelocityDeterminable;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.math.Vec3d;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+
+@Mixin(Entity.class)
+public class EntityMixin implements VelocityDeterminable {
+    @Unique
+    public Vec3d prevPos = Vec3d.ZERO;
+    public Vec3d motion = Vec3d.ZERO;
+
+    @Override
+    public void storePrevPos() {
+        Vec3d pos = ((Entity)((Object)this)).getPos();
+        motion = pos.subtract(prevPos);
+        prevPos = pos;
+    }
```

Comment:
I don't think the velocity spiking is necessarily bad behavior, but you could store and compare the world here, setting the velocity back to 0 if the dimension is different to the previous tick. Something like this might work:

```suggestion
    private RegistryKey<World> prevWorld;

    @Override
    public void storePrevPos() {
        var entity = (Entity) (Object) this;
        var pos = entity.getPos();
        var world = entity.getWorld().getRegistryKey();
        
        // Avoid velocity spikes when changing dimensions
        if (prevWorld != world) {
            prevWorld = world;
            motion = Vec3d.ZERO;
        } else {
            motion = pos.subtract(prevPos);        
        }
        
        prevPos = pos;
    }
```

Double check this still works with tickPassenger. I believe it will; the prevPos gets observed twice but the world has already changed by the time tickPassenger is called.

Reply:
Actually, entities that have a passenger suppress the dimension change for both itself and the passenger. The one exception to this that I can think of is parrots, but they:
a) aren't a passenger in the same way that mountable entities are
b) don't get detected by the sensor anyways.

e20bcd3 adds the world-change velocity reset though, it has been tested and works perfectly.